compiler/noirc_frontend/src/parser/parser/types.rs

```diff
@@ -370,15 +370,15 @@ impl Parser<'_> {
     }
 
     fn parses_mutable_reference_type(&mut self) -> Option<UnresolvedTypeData> {
-        if self.eat(Token::Ampersand) {
-            if !self.eat_keyword(Keyword::Mut) {
-                self.expected_mut_after_ampersand();
-            }
+        // The `&` may be lexed as a slice start if this is an array or slice type
+        if self.eat(Token::Ampersand) || self.eat(Token::SliceStart) {
```

Comment:
I don't understand this. If we get "&[" then we also try to eat "mut" next, so we'd get "&[ mut" which isn't valid syntax.

Reply:
`&[` would be `SliceStart` followed by a `LeftBracket`